### PR TITLE
Add: Forte's `wiki_parser` example running on the Argo backend

### LINKED ISSUE #20: Implement the `wiki_parser` Forte example using the Argo backend.
In order to demonstrate that `PIRlib` can handle serialization of complex functions and that the `Argo` backend can easily work with sufficiently large amount of data present in NFS, an example needs to be added.

The `wiki_parser` example in **Forte** is a good choice for this because:

1. It deals with reading and writing data to and from disk.
2. There are multiple, sequentially dependent stages of data processing which form a pipeline.
3. **Forte** is part of the **CASL** framework and hence it would also prove that different **CASL** components can work hand-in-hand.

@hunterhector @zhanyuanucb

### LINKED ISSUE #22: Support for specifying input files in the Argo backend
Currently, only NFS directories can be specified as inputs to the Argo Workflows. The `inproc` and the `docker` backend both have the support for files (stored in an NFS) to be specified as input.

The Argo backend can be updated to have support for input files along with directories.

### PR DESCRIPTION
**Pull request by Nilabhra**
This PR introduces the following additions/changes:

- Adds the `wiki_parser` example which runs on the Argo backend.
- Minor bug fixes in the `dockerize` module.
- Adds support for specifying input files stored in NFS to the Argo backend.

Closes #20
Closes #22

Screenshot 2022-12-27 at 1 11 40 PM

### CODE REVIEW

**Comment by zhanyuanucb** [comment]
I noticed that there is a lot of repeated code in `examples/wiki_parser/full_pipeline.py` and `examples/wiki_parser/sample_pipeline.py`. Any chance we can avoid this?

File: docs/README.rst [hunk 1]
```diff
@@ -69,16 +69,44 @@ The following steps require a existing docker installation.
...
+Follow the instructions `here <https://argoproj.github.io/argo-workflows/quick-start/>`_ and navigate your browser to ``https://127.0.0.1:2746``.
+
+Finally, execute the example.
+::
+
+   bash examples/multi_backends/run_argo.sh
+
+
+You should be able to see the live execution of the different steps of the pipeline in the browser.
+Open up ``example/package_argo.yml``, ``argo-train.yml`` and see what's inside.
```
[the remark this answers is not in the document]
Comment (Nilabhra) [reply docs/README.rst:108]: Thanks for noticing!

**Comment by Nilabhra** [comment]
@zhanyuanucb 
> I noticed that there is a lot of repeated code in `examples/wiki_parser/full_pipeline.py` and `examples/wiki_parser/sample_pipeline.py`. Any chance we can avoid this?
I can definitely modularize the `add_wiki_info` and `read_wiki_text`. Would that be fine? 

I think it's also possible to modularize the other `@task` functions, but in that case, the user would have to pass the names of the `tql` and `tql.bz` files as env vars when running the `dockerize` module. I kept them hard coded in the code because it makes the examples a bit more transparent. Let me know what you would prefer.

File: docs/README.rst [hunk 2]
```diff
@@ -69,16 +69,44 @@ The following steps require a existing docker installation.
...
 a docker-compose workflow from the serialized pipeline and save it to
 ``example/docker-compose.yml``, (3) execute the generated docker-compose
 workflow.
 
-Open up ``example/run_docker.sh``, ``example/package_docker.yml``, and
-``example/docker-compose.yml`` and see what's inside.
+Open up ``examples/multi_backends/run_docker.sh``, ``example/multi_backends/package_docker.yml``, and
+``examples/multi_backends/docker-compose.yml`` and see what's inside.
+
+Running as an Argo Workflow
```
[the remark this answers is not in the document]
Comment (Nilabhra) [reply docs/README.rst:86]: Indeed it was. Fixed now!

File: docs/README.rst [hunk 3]
```diff
@@ -69,16 +69,44 @@ The following steps require a existing docker installation.
...
 
 This script will (1) automatically dockerize the local environment and
 serialize the pipeline into ``example/package_docker.yml``, (2) generate
 a docker-compose workflow from the serialized pipeline and save it to
 ``example/docker-compose.yml``, (3) execute the generated docker-compose
 workflow.
 
-Open up ``example/run_docker.sh``, ``example/package_docker.yml``, and
-``example/docker-compose.yml`` and see what's inside.
+Open up ``examples/multi_backends/run_docker.sh``, ``example/multi_backends/package_docker.yml``, and
```
[the remark this answers is not in the document]
Comment (Nilabhra) [reply docs/README.rst:83]: Thanks for noticing. Fixed now.

File: docs/README.rst [hunk 4]
```diff
@@ -101,14 +104,73 @@ Follow the instructions `here <https://argoproj.github.io/argo-workflows/quick-s
...
+
+If you are executing the workflow on the full data without first executing on the sample data, do the following:
+
+::
 
+   $ mkdir examples/wiki_parser/outputs
+   $ bash examples/wiki_parser/run_full_pipeline.sh
 
-.. include-end-before
+The outputs would appear in the outputs/ directory and the Workflow execution can be viewed from the br
```
Comment (zhanyuanucb) [inline docs/README.rst:173]: what is br?

Comment (Nilabhra) [reply docs/README.rst]: Typo, should have been browser. I guess parts of it got deleted somehow. Fixed it.

**Review by zhanyuanucb — APPROVED** [review]

**Pull request merged**